backend/src/main/java/com/ethan5/controller/FriendController.java

```diff
@@ -1,21 +1,38 @@
 package com.ethan5.controller;
 
 import com.ethan5.dto.FriendRequest;
+import com.ethan5.entity.Friend;
 import com.ethan5.service.FriendService;
 import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.List;
+
 @RestController
 @AllArgsConstructor
 @RequestMapping("v1/friend")
 public class FriendController {
     private final FriendService service;
 
-    @PostMapping
-    public void addFriend(@RequestBody FriendRequest req) {
-        service.addFriend(req);
+    @PostMapping("{id}/{id2}")
+    public void addFriend(@PathVariable("id") String id1,
```

Comment:
Quick question: here, we de-structured the FriendRequest record only to restructure it again. Why are we doing that?

Reply:
It made the request nicer to build from the frontend, this can be considered temporary as we haven't solidified any of our request parameter formats--in the future I want to add something like retrofit to the frontend and agree on a standard to use.